### DISCUSSION TITLE
🐞 Dagger is generating malformed Go code

### DISCUSSION THREAD

**Discussion by gtabrar** [dagger/dagger#12187] [Help]
### What is the issue?

This Go code in a dagger module:
```
func (m *Rack) Join(r *Rack) *Rack {
    m.Name = m.Name + "+" + r.Name
    m.Services = append(m.Services, r.Services...)
    return m
}
```

Gets generated as:
```
func (r *Rack) Join(r *Rack) *Rack {          
        assertNotNil("r", r)
        q := r.query.Select("join")           
        q = q.Arg("r", r)
                                             
        return &Rack{
                query: q,                   
        }   
} 
```

This fails on the redeclaration of `r`—if I change my input variable to anything other than r, it builds fine.

### Dagger version

dagger v0.13.5 (registry.dagger.io/engine:v0.13.5) darwin/arm64

### Steps to reproduce

1. Create a module using the code block above
2. Attempt to install that module from another module

### Log output

```
Stderr:
# dagger/rack-example/internal/dagger
internal/dagger/dagger.gen.go:7017:21: r redeclared in this block
        internal/dagger/dagger.gen.go:7017:7: other declaration of r
```

**Comment by jedevc** [comment]
I think this is a specific case of https://github.com/dagger/dagger/issues/6186 - linking against that one.
